sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -20,247 +19,445 @@
+For example:
 
-We introduce a new function `get_wi_data` that provides a view of the portion of the matrix that is owned by the current WI. The indexing provided inside the `wi_data` class accesses only the portion of the current WI and returns  `wi_element`. This latter holds a reference to the original joint_matrix that `wi_data` was constructed from. This means that modifying `wi_data` also modifies the corresponding joint matrix elements. Users can use the `=` operator to update the element of the `joint_matrix` represented by the `wi_element` after the element-wise operation.
+```c++
+joint_matrix<sub_group, precision::tf32, use::a, tM, tK,
+             layout::row_major> tA;
+```
 
-Using `get_wi_data`, it is not possible to know which portions of data are owned by each thread in the group as this is implementation defined and changes from one backend to the other. For general piece-wise operations such as summing the rows of a matrix, the WI data to joint matrix mapping coordinates information must be known in order to reason about the matrix view and extract the relevant piece. However, for element-wise operations where the same operation is performed on all the elements of the matrix, having all the WIs in the group apply the operation inside a loop iterating over the `length` of `wi_data` guarantees the whole matrix element-wise operation.   
+Whenever the application loads, stores, fills, or accesses the
+elements of a TF32 matrix, the application sees the elements as
+float. There are special overloads of these functions for TF32 for
+this purpose.
 
-Therefore, this extension currently only supports class 1 of operations because the mapping between `get_wi_data` and `joint_matrix` elements is not required to be known for these operations. However, general piece-wise operations will be supported in the future as a new API will be provided to convey the mapping from `joint_matrix` domain to WI Domain (See Section "WI data to joint matrix mapping coordinates information for piece-wise operations for more information").
+==== TF32 load
+These overloads of `joint_matrix_load` load float values into a TF32
+matrix. It is unspecified whether the implementation loads all 32 bits
+into the joint matrix or if it only loads the relevant 19 bits.
 
-Also, note that `get_wi_data` cannot return a fixed size array length because the length of the WI portion is a runtime variable for the following reasons:
+```c++
+namespace sycl::ext::oneapi::experimental::matrix {
 
-1- The main compilation mode of SYCL is JIT compilation and partitioning among WIs is implementation defined.
+template <typename Group, size_t Rows, size_t Cols,
+          access::address_space Space, access::decorated IsDecorated>
+void joint_matrix_load(Group g,
+    joint_matrix<Group, precision::tf32, use::accumulator, Rows, Cols,
+    layout::dynamic> &res,
+    multi_ptr<float, Space, IsDecorated> src, size_t stride, layout Layout);
```

Comment:
```suggestion
    multi_ptr<const float, Space, IsDecorated> src, size_t stride, layout Layout);
```

Keep this overload in sync with the main definition above.  We decided recently that the `multi_ptr` should be `const` for loads, so do that here too.  Same for the other overload below.